Python/Product/Analysis/Analyzer/InterpreterScope.cs

```diff
@@ -278,7 +246,7 @@ public InterpreterScope(AnalysisValue av, InterpreterScope outerScope)
         }
 
         internal virtual void ClearVariables() {
-            _variables.Clear();
+            _variables = new AnalysisDictionary<string, VariableDef>();
```

Comment:
`_variables` is directly referenced in cloning constructor. Is it affected?

Reply:
Same thing with `_nodeScopes`

Reply:
This isn't a problem because we only modify the fields on the analysis thread. On any query threads, we shouldn't be relying on reference matches between the contents of these dictionaries.